src/handformer.cpp

```diff
@@ -63,12 +72,89 @@ auto Mahjong::HandFromNotation(std::string notation) -> std::vector<Piece> {
         if (isError || i == -1) {
           return Piece(Piece::ERROR);
         }
+
         if (isHonor) {
-          return Piece(NOTATION_TO_HONOR[i]);
+          return Piece(NOTATION_TO_HONOR.at(i));
         }
-        return Piece::formPiece(setSuit, i);
+
+        auto isRedFive = i == 0;
+        auto value = isRedFive ? 5 : i;
+
+        return Piece::formPiece(setSuit, value, isRedFive);
       });
     currentTiles.clear();
   }
+  // Free Remaining tiles (in case of no ending suit)
+  currentTiles.clear();
+
```

Comment:
Shouldn't you return an error piece or some kind of error? Or are we expecting a user to call isValid before expecting valid output from the this function?

Reply:
comment added in 5ee5d6df3ae88121f47d7fc03d03b9139bcb821f

It is the users responsibility to verify their input, HandFromNotation will regardless convert what it can